apps/dav/lib/Connector/Sabre/Server.php

```diff
@@ -47,20 +48,17 @@ public function start() {
 			$this->httpRequest->setBaseUrl($this->getBaseUri());
 			$this->invokeMethod($this->httpRequest, $this->httpResponse);
 		} catch (\Throwable $e) {
-			if ($e instanceof \TypeError) {
+			try {
+				$this->emit('exception', [$e]);
+			} catch (\Exception $ignore) {
+			}
+
+			if ($e instanceof TypeError) {
 				/*
 				 * The TypeError includes the file path where the error occurred,
 				 * potentially revealing the installation directory.
-				 *
-				 * By re-throwing the exception, we ensure that the
-				 * default exception handler processes it.
 				 */
-				throw $e;
-			}
-
-			try {
-				$this->emit('exception', [$e]);
-			} catch (\Exception $ignore) {
+				$e = new TypeError('A type error occurred. For more details, please refer to the logs, which provide additional context about the type error.');
```

Comment:
```suggestion
				$e = new TypeError('A type error occurred. For more details, please refer to the logs, which provide additional context about the type error.', 0, $e);
```

Would it be a problem to store the previous exception? It seems it’s only part of the response in debug mode.

Reply:
Good question :+1: 

The debugExceptions blocks appear to be dead code because there's no code in server setting it to true. Sounds like a nice enhancement to also include a stack trace, when debug enabled, for dav requests.

For non-dav requests we already include the stack trace with debug true.